openssl/openssl.php

```diff
@@ -542,18 +544,114 @@ function openssl_pkcs12_read(string $pkcs12, &$certificates, string $passphrase)
+ * previously generated by <b>openssl_pkey_new</b> (or
+ * otherwise obtained from the other openssl_pkey family of functions).
+ * The corresponding public portion of the key will be used to sign the
+ * CSR.
+ * </p>
+ * @param array|null $options [optional] <p>
+ * By default, the information in your system openssl.conf
+ * is used to initialize the request; you can specify a configuration file
+ * section by setting the config_section_section key of
+ * <i>configargs</i>. You can also specify an alternative
+ * openssl configuration file by setting the value of the
+ * config key to the path of the file you want to use.
+ * The following keys, if present in <i>configargs</i>
+ * behave as their equivalents in the openssl.conf, as
+ * listed in the table below.
+ * <table>
+ * Configuration overrides
+ * <tr valign="top">
+ * <td><i>configargs</i> key</td>
+ * <td>type</td>
+ * <td>openssl.conf equivalent</td>
+ * <td>description</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>digest_alg</td>
+ * <td>string</td>
+ * <td>default_md</td>
+ * <td>Selects which digest method to use</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>x509_extensions</td>
+ * <td>string</td>
+ * <td>x509_extensions</td>
+ * <td>Selects which extensions should be used when creating an x509
+ * certificate</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>req_extensions</td>
+ * <td>string</td>
+ * <td>req_extensions</td>
+ * <td>Selects which extensions should be used when creating a CSR</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>private_key_bits</td>
+ * <td>integer</td>
+ * <td>default_bits</td>
+ * <td>Specifies how many bits should be used to generate a private
+ * key</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>private_key_type</td>
+ * <td>integer</td>
+ * <td>none</td>
+ * <td>Specifies the type of private key to create. This can be one
+ * of <b>OPENSSL_KEYTYPE_DSA</b>,
+ * <b>OPENSSL_KEYTYPE_DH</b> or
+ * <b>OPENSSL_KEYTYPE_RSA</b>.
+ * The default value is <b>OPENSSL_KEYTYPE_RSA</b> which
+ * is currently the only supported key type.
+ * </td>
+ * </tr>
+ * <tr valign="top">
+ * <td>encrypt_key</td>
+ * <td>boolean</td>
+ * <td>encrypt_key</td>
+ * <td>Should an exported key (with passphrase) be encrypted?</td>
+ * </tr>
+ * <tr valign="top">
+ * <td>encrypt_key_cipher</td>
+ * <td>integer</td>
+ * <td>none</td>
+ * <td>
+ * One of cipher constants.
+ * </td>
+ * </tr>
+ * </table>
+ * </p>
+ * @param array|null $extra_attributes [optional] <p>
+ * <i>extraattribs</i> is used to specify additional
+ * configuration options for the CSR. Both <i>dn</i> and
+ * <i>extraattribs</i> are associative arrays whose keys are
+ * converted to OIDs and applied to the relevant part of the request.
+ * </p>
+ * @return OpenSSLCertificateSigningRequest|false the CSR.
+ */
+function openssl_csr_new(array $distinguished_names, &$private_key, ?array $options, ?array $extra_attributes): OpenSSLCertificateSigningRequest|false
+{ }
```

Comment:
LanguageLevelTypeAware also can be used here for return type as this function is completely equal to above one except return type. Do I miss something?

Reply:
Yep, fixed. :+1: